managers/bosh-manager/DirectorService.js

```diff
@@ -778,7 +769,14 @@ class DirectorService extends BaseDirectorService {
     return Promise.join(
         this.executeActions(CONST.SERVICE_LIFE_CYCLE.PRE_BIND, actionContext),
         this.getDeploymentIps(deploymentName),
-        (preBindResponse, ips) => this.agent.createCredentials(ips, binding.parameters, preBindResponse)
+        (preBindResponse, ips) => utils.retry(() => this.agent.createCredentials(ips, binding.parameters, preBindResponse), {
+          maxAttempts: 3,
+          timeout: Math.floor(config.http_timeout * 2 / 3)
```

Comment:
Ok so the timeout is configured here. The problem could be if the first IP is not reachable and taking time, always it would time out. I was thinking of timeout per request.

Reply:
Sure. Will do it.